### DISCUSSION TITLE
How to configure the heartbeat duration/deadline/time for leader election performed using client/v3?

### DISCUSSION THREAD

**Discussion by shreemaan-abhishek** [etcd-io/etcd#14655] [Q&A] [answered]
I was using [this](https://gist.github.com/shreemaan-abhishek/e612fc9f52c2d786fc666da47765b8a5) sample program to understand how leader election happens under the hood in etcd when using the etcd client. I build and run the program in two separate shell instances. One of the instances becomes the leader and executes whereas the other one blocks.

When I interrupt the leader instance using ^C, the program execution ends but the lease is still held by the dead program. Ideally, the lease object should vanish as soon as the active process ends. By default, the time taken by the lease object to vanish away is too long. How do I make it shorter?

Here's a video of the demo: https://www.youtube.com/watch?v=qqr1MXOLizI

**Comment by tjungblu** [comment, marked as answer]
The session has TTL and lease arguments for such crash scenarios:
https://github.com/etcd-io/etcd/blob/main/client/v3/concurrency/session.go#L113-L141

**Reply by shreemaan-abhishek** [reply]
Wow, thank you so much sire! Thanks for taking out the time to help out a random stranger. I promise I will carry your gesture as a legacy ❤️
